micromagneticdata/abstract_drive.py

```diff
@@ -316,8 +317,19 @@ def to_xarray(self, *args, **kwargs):
         if len(self._step_files) == 1:
             darray = self[0].to_xarray(*args, **kwargs)
         else:
-            field_darrays = (field.to_xarray(*args, **kwargs) for field in self)
-            darray = xr.concat(field_darrays, dim=self.table.data[self.table.x])
+            # xr.stack (below) is too slow and needs a lot of memory
+            # field_darrays = (field.to_xarray(*args, **kwargs) for field in self)
+            # darray = xr.concat(field_darrays, dim=self.table.data[self.table.x])
+            array = np.empty(
+                (self.n, *self[0].mesh.n, self.m0.dim), dtype=self[0].array.dtype
+            )
+            for i, field in enumerate(self):
+                array[i] = field.array
+            coords = dict(self[0].to_xarray().coords)
```

Comment:
Did you try keeping just `coords = self[0].to_xarray().coords`? It might work as is.

Reply:
I've just tried it but it doesn't work (complains about new coordinates without associated dimensions). I'll keep the dictionary conversion.